### PR TITLE
Filtering categories by languade

### PR DESCRIPTION
**Pull request by toslan**
Fix issue described here: https://github.com/joomla/joomla-cms/issues/5734

### CODE REVIEW

**Comment by toslan** [comment]
I suggest this patch to solve issues 
https://github.com/joomla/joomla-cms/issues/5734
https://github.com/joomla/joomla-cms/issues/5740

File: libraries/legacy/categories/categories.php [hunk 1]
```diff
@@ -241,6 +241,11 @@ protected function _load($id)
 		{
 			$query->where('c.published = 1');
 		}
+// Filter category by language
+		if (JLanguageMultilang::isEnabled())
+		{
+			$query->where('c.language in (' . $db->quote(JFactory::getLanguage()->getTag()) . ',' . $db->quote('*') . ')');
+		}
```
Comment (JoomliC) [inline libraries/legacy/categories/categories.php]: Remove the tab after } for Travis to succeed ;-)

File: libraries/legacy/categories/categories.php [hunk 2]
```diff
@@ -241,6 +241,11 @@ protected function _load($id)
 		{
 			$query->where('c.published = 1');
 		}
+// Filter category by language
```
Comment (zero-24) [inline libraries/legacy/categories/categories.php]: @toslan please add tabs here to agline it :smile:

Comment (JoomliC) [inline libraries/legacy/categories/categories.php]: and an enter before :smiley:

Comment (zero-24) [inline libraries/legacy/categories/categories.php]: >and an enter before 

:+1: hehe but this please without tabs :smiley:

**Comment by JoomliC** [comment]
@test success!

Before PR, categories were not filtered by language.

PR tested with multi-language enabled, and with those settings : 
- CAT_LANG1
 - - SUB_CAT_LANG1
 - - SUB_CAT_LANG2
  - - - SUB_SUB_CAT_LANG1
 - - SUB_CAT_LANG3
- CAT_LANG2
 - - SUB_CAT_LANG1
 - - SUB_CAT_LANG3

Menus (Type: category list or category blog) : 
1. Menu LANG_1 -> CAT_LANG1
2. Menu LANG_2 -> CAT_LANG2
3. Menu LANG3 -> CAT_LANG2

Results with PR : 
1. CAT_LANG1 + SUB_CAT_LANG1
2. CAT_LANG2 (no subcategory displayed each time a parent is not in current language)
3. error 404 "Category not found"

So, all is good! :+1: 
Note that about 3, i will check this week-end (or if another person is looking at it already...) if possible to add a checking when saving a menu, to prevent to be able to save a menu if menu language is not the same as the category's language selected.

Cyril

File: libraries/legacy/categories/categories.php [hunk 3]
```diff
@@ -241,6 +241,11 @@ protected function _load($id)
 		{
 			$query->where('c.published = 1');
 		}
+		// Filter category by language
```
Comment (JoomliC) [inline libraries/legacy/categories/categories.php]: I will just add here before a new line, to separate the filter by language from the filter by state ;-)

**Comment by infograf768** [comment]
Will test later.

**Comment by zero-24** [comment]
New PR by @toslan https://github.com/joomla/joomla-cms/issues/5817

**Comment by toslan** [comment]
I came across with another issue related to category language filter. The Language switcher module uses path helper to compute associated URLs. Since inactive languages are now filtered out the URLs for category views cannot be computed this way. Moreover path helper always loads complete category views for all languages which is wasteful because users usually do not switch to different languages while browsing.
I suggest PR https://github.com/joomla/joomla-cms/pull/5817 to fix this issue. It uses patch https://github.com/joomla/joomla-cms/pull/5756 (which is now closed) and some more files patched. No more category path helper called. Category objects for inactive languages are not loaded anymore.

**Pull request closed without merging**